src/styles/blocks/container.scss

```
@@ -0,0 +1,35 @@
+.container {
+  display: grid;
+
+  @for $i from 1 through 6 {
+    & :nth-child(#{$i})::after {
+      content: "#{$i}";
+    }
+  }
+
+  $colors-set:
+    1 "rgb(255, 0 , 0)",
+    2 "rgb(204, 0 , 0)",
+    3 "rgb(153, 0 , 0)",
+    4 "rgb(102, 0 , 0)",
+    5 "rgb(51, 0 , 0)",
+    6 "rgb(0, 0 , 0)";
+
+  @each $num, $color in $colors-set {
+    & :nth-child(#{$num}) {
+      background-color: #{$color};
+    }
+  }
+  
+  @media (min-width: 600px) and (max-width: 899px) {
+    & {
+      grid-template-columns: repeat(2, minmax(300px, 1fr));
+    }
+  }
+  
+  @media(min-width: 900px) {
+    & {
+      grid-template-columns: repeat(3, minmax(300px, 1fr));
+    }
+  }
```

You can do it with one media require. For example: 
```suggestion

  @media (min-width: 900px) {
    grid-template-columns: 1fr 1fr 1fr;
  }
```
But before you need to use autof-ill with minmax()